crates/sui-framework/packages/dwallet-system/sources/dwallet.move

```diff
@@ -454,6 +455,49 @@ module dwallet_system::dwallet {
         encryption_key_id: ID,
     }
 
+    /// Owned object that holds all encrypted user shares that were sent to the user,
+    /// and have been verified, re-signed, and re-encrypted by the user.
+    /// `encrypted_user_shares` is a key-value table where the key is the Dwallet ID and the value is the EncryptedUserShare object ID.
+    struct EncryptedUserShares has key {
+        id: UID,
+        encrypted_user_shares: Table<ID, ID>,
+    }
+
+    public fun create_encrypted_user_shares(ctx: &mut TxContext) {
+        let holder = EncryptedUserShares {
+            id: object::new(ctx),
+            encrypted_user_shares: table::new(ctx),
+        };
+        transfer::transfer(holder, tx_context::sender(ctx));
+    }
+
+    public fun save_encrypted_user_share(
+        encrypted_user_shares: &mut EncryptedUserShares,
+        encrypted_user_share: &EncryptedUserShare,
+        encryption_key: &EncryptionKey,
+        ctx: &mut TxContext
+    ) {
+        // TODO: add signature verification
```

Comment:
```suggestion
```
What signature needs to be verified here? I believe only when the `EncryptedUserShare` object is created it should be verified